### PR TITLE
Deprecate and warn about --genscript, fixes #730

### LINKED ISSUE #730: deprecating --genscript
Originally reported by: **Ronny Pfannschmidt (BitBucket: [RonnyPfannschmidt](http://bitbucket.org/RonnyPfannschmidt), GitHub: [RonnyPfannschmidt](http://github.com/RonnyPfannschmidt))**

----------------------------------------

the genscript method for distributing pytest seems to be used not that widely,
doesnt really support plugins and googleing for it reveals a lot of issues people have

meanwhile putting apps into zips became much better

https://www.python.org/dev/peps/pep-0441/

----------------------------------------
- Bitbucket: https://bitbucket.org/pytest-dev/pytest/issue/730

### PR DESCRIPTION
**Pull request by cpommranz**

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,6 @@
 2.8.0.dev (compared to 2.7.X)
 -----------------------------
+- deprecate and warn about the --genscript option
```
Comment (flub) [inline CHANGELOG]: You should add something like "Thanks cpommranz for the PR" or something (or use your real name)

**Comment by astrofrog** [comment]
@cpommranz @flub - could you provide more information about the decision behind this? Is there any replacement or is the idea to get people to install py.test by hand now?

**Comment by RonnyPfannschmidt** [comment]
genscript could never propperly support pytest plugins to begin with, also tooling is much better now, and one can build a zipped application of pytest (which will be documented before removal)

**Comment by nicoddemus** [comment]
Could we put those reasons into the documentation as well? I'm sure people who uses it and don't follow pytest's repo will also wonder why the deprecation.

**Pull request merged**